### PR TITLE
Promote GetNewBlockV3 and enable by default

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This PR unhides the GetNewBlockV3 and enables it by default.

## Fixed Issue(s)

Part of #7779 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by tbenr — COMMENTED** [review]
Don't know if we really want to expose the `--block-v3-enabled`.. if we do that we theoretically add it in docs and it will be a problem when we deprecate the option completely (cose it will be a breaking change). I might be wrong, just posing the questions :)

**Comment by gfukushima** [comment]
Make sense to me, I'll keep it as experimental them.

File: teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorOptions.java [hunk 1]
```diff
@@ -139,10 +139,9 @@ public class ValidatorOptions {
   private int executorThreads = DEFAULT_VALIDATOR_EXECUTOR_THREADS;
 
   @Option(
-      names = {"--Xblock-v3-enabled"},
+      names = {"--block-v3-enabled"},
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorOptions.java]: i think leaving it as a dev option thats just enabled by default was what i was thinking, i am hoping that this flag disappears once we prove we don't require it...

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,5 +23,6 @@
 - When `p2p-subscribe-all-subnets-enabled`, `p2p-peer-lower-bound` now defaults to 60 (previously 64), and `p2p-peer-upper-bound` now defaults to 80 (previously 100).
 - When `p2p-subscribe-all-subnets-enabled`,  (`Xnetwork-async-p2p-max-queue`, `Xnetwork-async-beaconchain-max-queue`, `Xp2p-batch-verify-signatures-queue-capacity`)  now default to 40_000 (previously 10_000)
 - When `p2p-subscribe-all-subnets-enabled`,  `Xvalidator-executor-max-queue-size`  now defaults to 60_000 (previously 40_000).
+- Enable [GetNewBlockV3](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/ValidatorRequiredApi/produceBlockV3) to default.
```
Comment (lucassaldanha) [inline CHANGELOG.md]: "by default"?

**Review by lucassaldanha — APPROVED** [review]

**Pull request merged**